platform/ext/target/nordic_nrf/nrf5340/spm_hal.c

```diff
@@ -46,9 +48,49 @@ enum tfm_plat_err_t tfm_spm_hal_configure_default_isolation(
                   uint32_t partition_idx,
                   const struct tfm_spm_partition_platform_data_t *platform_data)
 {
-    /* FIXME we need to see how to handle this because nRF IDAU peripheral
-     * is not able to assign PRIV/nPRIV access policy configuration
+#if defined(CONFIG_TFM_ENABLE_MEMORY_PROTECT) && (TFM_LVL != 1)
+    bool privileged = tfm_is_partition_privileged(partition_idx);
+    struct mpu_armv8m_region_cfg_t region_cfg;
+#endif
+
+    if (!platform_data) {
+        return TFM_PLAT_ERR_INVALID_INPUT;
+    }
+
+#if defined(CONFIG_TFM_ENABLE_MEMORY_PROTECT) && (TFM_LVL != 1)
+    if (!privileged) {
+        region_cfg.region_nr = PARTITION_REGION_PERIPH_START +
+                                periph_num_count;
+        periph_num_count++;
+        if (periph_num_count >= PARTITION_REGION_PERIPH_MAX_NUM) {
+            return TFM_PLAT_ERR_MAX_VALUE;
+        }
+        region_cfg.region_base = platform_data->periph_start;
+        region_cfg.region_limit = platform_data->periph_limit;
+        region_cfg.region_attridx = MPU_ARMV8M_MAIR_ATTR_DEVICE_IDX;
+        region_cfg.attr_access = MPU_ARMV8M_AP_RW_PRIV_UNPRIV;
+        region_cfg.attr_sh = MPU_ARMV8M_SH_NONE;
+        region_cfg.attr_exec = MPU_ARMV8M_XN_EXEC_NEVER;
+
+        mpu_armv8m_disable(&dev_mpu_s);
+
+        if (mpu_armv8m_region_enable(&dev_mpu_s, &region_cfg)
+            != MPU_ARMV8M_OK) {
+            return TFM_PLAT_ERR_SYSTEM_ERR;
+        }
+        mpu_armv8m_enable(&dev_mpu_s, PRIVILEGED_DEFAULT_ENABLE,
+                          HARDFAULT_NMI_ENABLE);
+    }
+#endif /* defined(CONFIG_TFM_ENABLE_MEMORY_PROTECT) && (TFM_LVL != 1) */
+
+    /* Peripheral for a (secure) partition is configured as Secure.
+     * Note that the SPU does not allow to configure PRIV/nPRIV permissions
+     * for address ranges, including peripheral address space.
      */
+    if (platform_data->periph_base_addr != 0) {
+        spu_periph_configure_to_secure(platform_data->periph_base_addr);
```

Comment:
Curious: I see that fex. AN521 configures MPU *and* ppc_en_secure_unpriv. What does ppc_en_secure_unpriv do that the MPU doesn't?

Reply:
@oyvindronningstad this is a good question: I suspect that the PPC controller in ARM MPS2+ FPGAs has the power to assign privilege/un-privilege access permissions. Nordic SPU only assigns RWX permissions (ignoring the privilege level). 

In Cortex-m all access policies of this kind are implementing the AND combination rule.

So, to be able to access a peripheral from un-privileged level, both the MPU and PPC need to allow this.